### PR TITLE
Alternative Field Encoding for Types

### PR DESCRIPTION
**Pull request by Doreen-Schwartz**
## Changes

- Added the option to pass an alternative encoder (`informationWriter`) to each Field for `nistEncode`. 
- Added the option to pass an alternative decoder (`informationDecoder`) to each Field for `nistDecode`.

### CODE REVIEW

File: src/nistVisitor.ts [hunk 1]
```diff
@@ -216,6 +216,15 @@ interface VisitNistFileParams<
   recordVisitor?: NistRecordVisitor<D, T>;
   fieldVisitor?: NistFieldVisitor<D, T>;
 }
+
+export const getPerTotOptions = <
+  T extends NistFieldCodecOptions,
+  U extends NistRecordCodecOptions<T>,
+>(
```
Comment (ivosh) [inline src/nistVisitor.ts:223]: Please add a return type of this function.

**Comment by ivosh** [comment]
@Doreen-Schwartz The interface changes look good! Please proceed with the rest of the implementation.

**Comment by Doreen-Schwartz** [comment]
I've added in unit tests + the requested changes. I think it's also worth adding this feature somewhere in the README, but I wasn't sure where would be most appropriate.

File: src/nistEncode.ts [hunk 2]
```diff
@@ -79,6 +80,15 @@ const invokeFormatters = ({
   return success(undefined);
 };
 
+const defaultInformationWriter = (
+  informationItem: Exclude<NistInformationItem, undefined>,
+): Buffer => {
+  if (typeof informationItem === 'string') {
```
Comment (ivosh) [inline src/nistEncode.ts:86]: A doubt about this condition: in all contexts where `defaultInformationWriter` is used, the `informationItem` is always a string.
Please could you comment under which circumstances the `informationItem` could be something else than a `string`?
Is the condition perhaps not necessary, leading to a simplification?

**Comment by ivosh** [comment]
@Doreen-Schwartz The PR looks really good! Thank you for the unit tests.
Please can you address the review comment. Then I will merge the PR, add something in README and publish a new version.

File: src/nistEncode.ts (hunk 2 above)
Comment (Doreen-Schwartz) [reply src/nistEncode.ts]: You're correct, the condition is unnecessary as it is already established that the `informationItem` is  a `string` at that point. I've removed it

**Comment by Doreen-Schwartz** [comment]
@ivosh Thanks for merging the PR! When can I expect this feature to be released onto npm so it can be used in my project?

**Comment by ivosh** [comment]
I am currently working on upgrading dependencies and updating README.

**Pull request merged**